framework/source/class/qx/core/Object.js

```diff
@@ -230,15 +230,30 @@ qx.Class.define("qx.core.Object",
     /**
      * Load user defined data from the object
      *
-     * @param key {String} the key
-     * @return {Object} the user data
+     * @param key {String?} the key
+     *   If key is null or undefined, pass the user the entire map
+     * @param bCopy {Boolean?}
+     *   If bCopy is truthy pass the user a deep clone
+     * @return {Object}
+     *   Returns null if no user data exists or if the given key has
+     *   not been set, the user data map as an object if requested, or
+     *   the value of the user data with the given key
      */
-    getUserData : function(key)
+    getUserData : function(key, bCopy)
     {
-      if (!this.__userData) {
-        return null;
+      var data;
+      if (key == null) {
```

Comment:
If we care, since this is `==` rather than `===`, this will also match if key is `""` (the empty string) or `0` or `false`. The latter two aren't likely a problem. I'm not sure whether an empty string key is considered a valid key or instead should return the whole map. I'll approve anyway, but others can may consider this.

Reply:
What's about  (key === null)  ||  (key === undefined)  instead? Then it would be absolutely clear what you mean and will not break code if someone has used 0 as key.

Reply:
I think that's how I'd probably do it.

Reply:
And the check   if (!this.__userData) { is necesary, see faild test.